### PR TITLE
Octane fixes

### PR DESCRIPTION
**Pull request by radislavB**
[JENKINS-55329: Commit are not reported to Octane from multibranch job](https://issues.jenkins-ci.org/browse/JENKINS-55329)
[JENKINS-55364 : Improve octane test connection validation](https://issues.jenkins-ci.org/browse/JENKINS-55364)
[JENKINS-55367 : Uft discovery job is not working correctly in Jenkins 2.107.1+ because of Security hardening](https://issues.jenkins-ci.org/browse/JENKINS-55367)

### CODE REVIEW

File: src/main/java/com/microfocus/application/automation/tools/octane/configuration/ConfigurationValidator.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
+/*
```
Comment (grount) [inline src/main/java/com/microfocus/application/automation/tools/octane/configuration/ConfigurationValidator.java]: Was it only renamed?

File: src/main/java/com/microfocus/application/automation/tools/octane/events/SCMListenerOctaneImpl.java [hunk 2]
```diff
@@ -48,53 +38,22 @@
...
+        if (!skipScmEvent(run)) {
+            CIEvent scmEvent = CIEventFactory.createScmEvent(run, scm);
+            if (scmEvent != null) {
+                CIJenkinsServicesImpl.publishEventToRelevantClients(scmEvent);
+            }
+        }
+    }
+
+    private static boolean skipScmEvent(Run<?, ?> run) {
+        boolean isMultibranchJob = run.getParent().getParent() != null &&
```
Comment (grount) [inline src/main/java/com/microfocus/application/automation/tools/octane/events/SCMListenerOctaneImpl.java]: Is the first getParent can return null as well?

File: src/main/java/com/microfocus/application/automation/tools/octane/events/SCMListenerOctaneImpl.java [hunk 3]
```diff
@@ -48,53 +38,22 @@
...
+                run.getParent().getParent().getClass().getName().equals(JobProcessorFactory.WORKFLOW_MULTI_BRANCH_JOB_NAME);
```
Comment (grount) [inline src/main/java/com/microfocus/application/automation/tools/octane/events/SCMListenerOctaneImpl.java]: Same

File: src/main/java/com/microfocus/application/automation/tools/octane/configuration/ConfigurationValidator.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class ConfigurationValidator {
```
[the remark this answers is not in the document]
Comment (grount) [reply src/main/java/com/microfocus/application/automation/tools/octane/configuration/ConfigurationValidator.java]: Sounds like a good idea to me, what do you think?

File: src/main/java/com/microfocus/application/automation/tools/octane/model/CIEventFactory.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import hudson.model.Run;
+import hudson.scm.SCM;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+
+/**
+ * Factory for creating ci event
+ */
+public final class CIEventFactory {
```
[the remark this answers is not in the document]
Comment (grount) [reply src/main/java/com/microfocus/application/automation/tools/octane/model/CIEventFactory.java]: Same

File: src/main/java/com/microfocus/application/automation/tools/octane/events/SCMListenerOctaneImpl.java (hunk 2 above)
Comment (radislavB) [reply src/main/java/com/microfocus/application/automation/tools/octane/events/SCMListenerOctaneImpl.java]: No, first  parent of the run - is job itself, it  cannot be null

File: src/main/java/com/microfocus/application/automation/tools/octane/configuration/ConfigurationValidator.java (hunk 1 above)
Comment (radislavB) [reply src/main/java/com/microfocus/application/automation/tools/octane/configuration/ConfigurationValidator.java]: renamed and refactored

**Comment by grount** [comment]
Approved for test build

**Pull request merged**